tests/manager/manager-ctor-write_concern-005.phpt

```diff
@@ -0,0 +1,74 @@
+--TEST--
+MongoDB\Driver\Manager::__construct(): write concern options (64-bit wtimeoutms)
+--SKIPIF--
+<?php if (4 !== PHP_INT_SIZE) { die('skip Only for 32-bit platform'); } ?>
+--FILE--
+<?php
+
+$tests = [
+    ['mongodb://127.0.0.1/?wtimeoutms=2147483648', []],
+    ['mongodb://127.0.0.1/?w=2&wtimeoutms=2147483648', []],
```

Comment:
Looking at [`_mongoc_uri_build_write_concern()`](https://github.com/mongodb/mongo-c-driver/blob/1.15.1/src/libmongoc/src/mongoc/mongoc-uri.c#L1606), which I also linked above, I'm not sure how this is successfully parsed by libmongoc.

[`mongoc_uri_parse_options()`](https://github.com/mongodb/mongo-c-driver/blob/1.15.1/src/libmongoc/src/mongoc/mongoc-uri.c#L1173) splits the query string, but [`mongoc_uri_apply_options()`](https://github.com/mongodb/mongo-c-driver/blob/1.15.1/src/libmongoc/src/mongoc/mongoc-uri.c#L1011) does the real work of converting original string values into their final types. Since [`MONGOC_URI_WTIMEOUTMS`](https://github.com/mongodb/mongo-c-driver/blob/1.15.1/src/libmongoc/src/mongoc/mongoc-uri.c#L740) is considered an int32 option, I expect [`mongoc_uri_parse_int32()`](https://github.com/mongodb/mongo-c-driver/blob/1.15.1/src/libmongoc/src/mongoc/mongoc-uri.c#L800) will truncate it. Assuming this test is actually passing, I'm not sure why we'd see `int(2147483648)` in the expected output below (at least for the test cases using only the connection string).

FWIW, this probably highlights something that was overlooked in [CDRIVER-3087](https://jira.mongodb.org/browse/CDRIVER-3087). Perhaps that was intentional, as I believe Swift cared more about the WriteConcern struct than URI parsing.

Reply:
> FWIW, this probably highlights something that was overlooked in [CDRIVER-3087](https://jira.mongodb.org/browse/CDRIVER-3087).

I concur. FWIW, `int(2147483648)` is still a 32-bit value, but it assumes an unsigned int (with the usual conversion happening when using signed ints). I noticed this as well and changing to `int(4294967296)` (expectedly) causes the test to fail:
> Unsupported value for "wtimeoutms": "4294967296".

I've created [CDRIVER-3396](https://jira.mongodb.org/browse/CDRIVER-3396) to track this. Should be fixable in 1.15.2 as support for 64-bit `wtimeoutms` was only introduced in 1.15.

I'd hold off on working on this PR until we've fixed the underlying libmongoc issue.

Reply:
We just assigned [CDRIVER-3396](https://jira.mongodb.org/browse/CDRIVER-3396) to you this morning, so feel free to move on to it before this.